clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -872,27 +878,24 @@ PointerVariableConstraint::mkString(Constraints &CS,
     case Atom::A_NTArr:
       if (emitArraySize(ConstArrs, TypeIdx, K))
         break;
-      AllArrays = false;
-      // This additional check is to prevent fall-through from the array.
-      if (K == Atom::A_NTArr) {
-        // If this is an NTArray.
-        getQualString(TypeIdx, Ss);
+      // If this is an NTArray.
```

Comment:
I noticed that in the `A_Arr` case `getQualString` is called before `emitArraySize`, but the reverse is done here. ISTM that this could cause qualifier to be dropped when rewriting constant sized null terminated arrays. Think it might be worth fixing while your in this code?

Reply:
I don't think this problem can actually occur because [array levels can't have their own type qualifiers](https://en.cppreference.com/w/c/language/array#Qualifiers).  (Do you have an example?)  Still, resolving the inconsistency seems like a reasonable motivation in its own right.  Done.

Reply:
No example, it was just a guess from looking at the code. As you say, improved consistency makes it worthwhile anyways.